### ISSUE TITLE
_.random() has a small bias against certain integer results

### ISSUE THREAD

**Issue by Venryx** [lodash/lodash#5099]
According to [this article](https://www.pcg-random.org/posts/bounded-rands.html), the "FP Multiply" method used by lodash's `random()` function is not 100% unbiased -- some integers will show up (very slightly) more than others.

Line in question: https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/random.js#L70

Equivalent code in the article: (under "FP Multiply (Biased)")
```
static uint32_t bounded_rand(rng_t& rng, uint32_t range) {
    double zeroone = 0x1.0p-32 * rng();
    return range * zeroone;
}
```

> This approach is just as biased as the classic modulo approach, but the bias manifests itself differently. For example, if we were choosing numbers in the range [0..52), the numbers 0, 13, 26 and 39 would appear once less often than the others.

Basically, the number of floating point values representable is (usually) not a multiple of the number of integer results in the target range, meaning some integers have one fewer floating-point value that maps to them -- creating a small bias.

The article provides various approaches that can be used to remove that small bias.

**Comment by Venryx** [comment]
Here is an article specifically explaining why the approach above is biased can be seen here: https://pthree.org/2018/06/13/why-the-multiply-and-floor-rng-method-is-biased
